packages/expo-notifications/plugin/src/withNotificationsIOS.ts

```diff
@@ -1,11 +1,67 @@
-import { ConfigPlugin, withEntitlementsPlist } from '@expo/config-plugins';
+import {
+  ConfigPlugin,
+  withEntitlementsPlist,
+  IOSConfig,
+  withXcodeProject,
+} from '@expo/config-plugins';
+import fs from 'fs-extra';
+import path from 'path';
 
-export const withNotificationsIOS: ConfigPlugin<{ mode: 'production' | 'development' }> = (
+import { NotificationsPluginProps } from './withNotifications';
+
+type XcodeProject = any;
+
+export const withNotificationsIOS: ConfigPlugin<NotificationsPluginProps> = (
   config,
-  { mode }
+  { mode = 'development', sounds = [] }
 ) => {
-  return withEntitlementsPlist(config, config => {
+  config = withEntitlementsPlist(config, config => {
     config.modResults['aps-environment'] = mode;
     return config;
   });
+  config = withNotificationSounds(config, { sounds });
+  return config;
+};
+
+export const withNotificationSounds: ConfigPlugin<{ sounds: string[] }> = (config, { sounds }) => {
+  return withXcodeProject(config, config => {
+    setNotificationSounds(sounds, {
+      projectRoot: config.modRequest.projectRoot,
+      project: config.modResults,
+    });
+    return config;
+  });
 };
+
+/**
+ * Save sound files to the Xcode project root and add them to the Xcode project.
+ */
+export function setNotificationSounds(
+  sounds: string[],
+  { projectRoot, project }: { project: XcodeProject; projectRoot: string }
+): XcodeProject {
+  if (!Array.isArray(sounds)) {
+    throw new Error(
+      `Must provide an array of sound files in your app config, found ${typeof sounds}.`
+    );
+  }
+
+  const projectName = IOSConfig.XcodeUtils.getProjectName(projectRoot);
+
+  sounds.map((soundFileRelativePath: string) => {
+    const fileName = path.basename(soundFileRelativePath);
+    const sourceFilepath = path.resolve(projectRoot, soundFileRelativePath);
+    const destinationFilepath = path.join(IOSConfig.Paths.getSourceRoot(projectRoot), fileName);
+
+    fs.copyFileSync(sourceFilepath, destinationFilepath);
+    if (!project.hasFile(`${projectName}/${fileName}`)) {
+      project = IOSConfig.XcodeUtils.addResourceFileToGroup(
```

Comment:
this part is scary, maybe have a snapshot test on the pbxproj so you know that after you've tested it locally, the results don't change. Maybe add a comment that gives a bit more info on what's supposed to happen here so no one changes it.

Reply:
Do you have an example testing adding these files to the `pbxproj` file? Since the UUID for an added resource file changes every run, a snapshot test here is a little tough. It looks like [Locales](https://github.com/expo/expo-cli/blob/522b989a1c5bf6cb88ecdd7330dd50ae4fc99857/packages/config-plugins/src/ios/Locales.ts#L66-L71) and [Google](https://github.com/expo/expo-cli/blob/522b989a1c5bf6cb88ecdd7330dd50ae4fc99857/packages/config-plugins/src/ios/Google.ts#L69-L74) plugins do this as well, but the `pbxproj` changes aren't tested

. maybe could do a deep search of the `XcodeProject` object and search for the expected values, and make sure they are the only ones there?